bitbucket-pipelines.yml

```diff
@@ -1,17 +1,122 @@
+# These are yaml anchors, reused later
+x-bash-env-steps:
+  - &bash_env_export export BASH_ENV="$BITBUCKET_CLONE_DIR/.bashrc"
+  - &bash_env_source source $BASH_ENV
+
+definitions:
+  caches:
+    terminusplugins: ${TERMINUS_PLUGINS_DIR:-~/.terminus/plugins}
+
+# This is a larger yaml anchor, reused for each pipeline
+default_steps: &default_steps
+  - step:
+      name: Configure Environment Variables
+      services:
+        - docker
+      caches:
+        - docker
+        - terminusplugins
+      script:
+        - *bash_env_export
+        - export CI_BRANCH=${BITBUCKET_BRANCH:-master} && echo $CI_BRANCH
+        - export PR_NUMBER=$BITBUCKET_PR_ID && echo $PR_NUMBER
+        - export CI_PR_URL=https://bitbucket.org/$BITBUCKET_REPO_OWNER/$BITBUCKET_REPO_SLUG/pull-requests/$BITBUCKET_PR_ID && echo $CI_PR_URL
+        - export CI_BUILD_NUMBER=$BITBUCKET_BUILD_NUMBER && echo $CI_BUILD_NUMBER
+        - export CI_PROJECT_USERNAME=$BITBUCKET_REPO_OWNER && echo $CI_PROJECT_USERNAME
+        - export CI_PROJECT_REPONAME=$BITBUCKET_REPO_SLUG && echo $CI_PROJECT_REPONAME
+        - export CI_PROJECT_NAME=$BITBUCKET_REPO_FULL_NAME && echo $CI_PROJECT_NAME
+        - ./.ci/build/set-environment
```

Comment:
using local set-environment script here, so that we can run build tools dev-master on top of arbitrary docker image

Reply:
> using local set-environment script here, so that we can run build tools dev-master on top of arbitrary docker image

Is that what you are doing or are you suggesting a change here?

Reply:
I would rather not maintain this set-environment file separately, but currently it's the only way to install build tools dev-master and work around bugs in docker image.

One of these needs to be resolved before we can get rid of set-environment from this repo:

  -  https://github.com/pantheon-systems/docker-build-tools-ci/pull/41  
  - https://github.com/pantheon-systems/docker-build-tools-ci/issues/36